components/mediation/event-sink/org.wso2.micro.integrator.event.sink/src/main/java/org/wso2/micro/integrator/event/sink/EventSink.java

```diff
@@ -0,0 +1,179 @@
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public String getReceiverUrlSet() {
+        return receiverUrlSet;
+    }
+
+    public void setReceiverUrlSet(String urlSet) {
+        this.receiverUrlSet = urlSet;
+    }
+
+    public String getAuthenticationUrlSet() {
+        return authenticationUrlSet;
+    }
+
+    public void setAuthenticationUrlSet(String urlSet) {
+        this.authenticationUrlSet = urlSet;
+    }
+
+    public DataPublisher getDataPublisher() {
+        return dataPublisher;
+    }
+
+    public void setDataPublisher(DataPublisher loadBalancingDataPublisher) {
+        this.dataPublisher = loadBalancingDataPublisher;
+    }
+
+    /**
+     * Generates an event sink from XML configuration element.
+     *
+     * @param eventSinkElement XML configuration element of event sink
+     * @param name             Name to be set for the created event sink
+     * @return Generated event sink
+     * @throws EventSinkException
+     */
+    public static EventSink createEventSink(OMElement eventSinkElement, String name) throws EventSinkException {
+
+        EventSink eventSink = new EventSink();
+
+        OMElement receiverUrl = eventSinkElement.getFirstChildWithName(EventSinkConstants.RECEIVER_URL_Q);
+        if (receiverUrl == null || "".equals(receiverUrl.getText())) {
+            throw new EventSinkException(
+                    EventSinkConstants.RECEIVER_URL_Q.getLocalPart() + " is missing in thrift endpoint config");
+        }
+        eventSink.setReceiverUrlSet(receiverUrl.getText());
+
+        OMElement authenticatorUrl = eventSinkElement.getFirstChildWithName(EventSinkConstants.AUTHENTICATOR_URL_Q);
+        if (authenticatorUrl != null) {
+            eventSink.setAuthenticationUrlSet(authenticatorUrl.getText());
+        }
+
+        OMElement userName = eventSinkElement.getFirstChildWithName(EventSinkConstants.USERNAME_Q);
+        if (userName == null || "".equals(userName.getText())) {
+            throw new EventSinkException(
+                    EventSinkConstants.USERNAME_Q.getLocalPart() + " is missing in thrift endpoint config");
+        }
+        eventSink.setUsername(userName.getText());
+
+        OMElement password = eventSinkElement.getFirstChildWithName(EventSinkConstants.PASSWORD_Q);
+        if (password == null || "".equals(password.getText())) {
+            throw new EventSinkException(
+                    EventSinkConstants.PASSWORD_Q.getLocalPart() + " attribute missing in thrift endpoint config");
+        }
+
+        try {
+            eventSink.setPassword(
+                    new String(CryptoUtil.getDefaultCryptoUtil().base64DecodeAndDecrypt(password.getText()),
+                            Charset.forName("UTF-8")));
+        } catch (CryptoException e) {
+            throw new EventSinkException("Failed to decrypt password", e);
+        }
+        eventSink.setName(name);
+
```

Comment:
Can we create the eventSink using the other constructor after all these? We will be able to remove the setters if we do that.

Reply:
Fixed in bd10bd3